server/assetstore.py

```diff
@@ -34,6 +34,43 @@
 DB_INFO_KEY = 'databaseMetadata'
 
 
+class DatabaseAssetstoreFile(dict):
+    """
+    This wraps a Girder file object dictionary so that the size parameter can
+    be determined only when actually necessary.
+    """
+    def __init__(self, file, adapter, *args, **kwargs):
+        """
+        Create a file dictionary where 'size' and 'mimeType' are determined
+        lazily.
+
+        :param file: the original file dictionary.
+        :param adapter: the owning assetstore adapter.
+        """
+        self._file = file
+        self._adapter = adapter
+        self._loaded = None
+        return super(DatabaseAssetstoreFile, self).__init__(file, *args, **kwargs)
+
+    def __getitem__(self, key, *args, **kwargs):
```

Comment:
We have two properties which are lazily evaluated.  Once evaluated, we just want to resolve to the default getitem.  To use `@property` would have required code like:

```
@property
def size(self):
  if not self._loaded:
    self._lazyload()
  return self.getter()
```

with this duplicated for `mimeType`, and the `_lazyload` function containing the bulk of what is in `__getitem__`.  I felt that the `__getitem__` implementation was simpler, and if I decided that another parameter would be lazy would be easier to extend (but you can argue that `@property` would be simpler).